### PR TITLE
[GPU] Replaced a part of cldnn::tensor usages

### PR DESCRIPTION
**Pull request by vladimir-paramuzov**
### Details:
 - Get rid of some cldnn::tensor usages for further replacement with ov::PartialShape

### CODE REVIEW

File: src/plugins/intel_gpu/src/graph/convolution.cpp [hunk 1]
```diff
@@ -382,21 +365,12 @@ convolution_inst::typed_primitive_inst(network& network, convolution_node const&
                               "expected output size",
                               1,
                               "Only one-dimensional batch size are supported");
-        CLDNN_ERROR_LESS_THAN(node.id(),
+        CLDNN_ERROR_NOT_EQUAL(node.id(),
```
Comment (lznamens) [inline src/plugins/intel_gpu/src/graph/convolution.cpp:368]: Are you sure that this is correct macro for all group cases (when group is equal to 1 and when group isn't equal to 1)? @sshlyapn what do you think?

File: src/plugins/intel_gpu/src/graph/deconvolution.cpp [hunk 2]
```diff
@@ -70,17 +60,14 @@ layout deconvolution_inst::calc_output_layout(deconvolution_node const& node) {
                                        0,
                                        "User-defined size of output layout must be positive (>= 1)");
 
-        tensor output_size(desc->output_size.batch[0],
+        tensor output_size(input_layout.batch(),
```
Comment (lznamens) [inline src/plugins/intel_gpu/src/graph/deconvolution.cpp:63]: Why not output batch?

Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/graph/deconvolution.cpp:63]: The value is the same, but the `output_size` parameter will be removed eventually

File: src/plugins/intel_gpu/src/graph/convolution.cpp [hunk 3]
```diff
@@ -382,21 +365,12 @@ convolution_inst::typed_primitive_inst(network& network, convolution_node const&
...
                               "Weights feature maps number",
-                              input_layout.size.feature[0],
+                              filter_inst.ifm() * filter_inst.group(),
                               "input feature maps number",
-                              weights_ifm,
+                              input_layout.feature(),
                               "Weights/ifm mismatch");
-
-        if (!argument.grouped_weights_shape && !format::is_grouped(filter_inst.format)) {
```
[the remark this answers is not in the document]
Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/graph/convolution.cpp:392]: it's covered by the check above

File: src/plugins/intel_gpu/src/graph/convolution.cpp (hunk 1 above)
Comment (vladimir-paramuzov) [reply src/plugins/intel_gpu/src/graph/convolution.cpp:368]: I think yes. At least all tests pass with such check

Comment (sshlyapn) [reply src/plugins/intel_gpu/src/graph/convolution.cpp:368]: As for me the condition looks correct

**Review by sshlyapn — APPROVED** [review]

**Review by lznamens — APPROVED** [review]

**Pull request merged**